### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by tfussell**
This PR initializes a new library which extracts the `application.giantswarm.io` group out of `apiextensions`. The dependence of `apiextensions` on `cluster-api` v0.3 means we can't import it in any projects that use `cluster-api` v0.4 or later. Splitting this API into a separate library means any operator or tool that requires these types can import them here without having to import `cluster-api`.

This library will be immediately used by `cluster-apps-operator` as part of the CAPO epic but should also be used by `app-operator`, `chart-operator`, etc in the future.

Towards https://github.com/giantswarm/giantswarm/issues/19402 and required for https://github.com/giantswarm/cluster-apps-operator/pull/118

See https://github.com/giantswarm/giantswarm/issues/19586 for more info on splitting `apiextensions`.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Review by rossf7 — COMMENTED** [review]
> This library will be immediately used by cluster-apps-operator as part of the CAPO epic but should also be used by app-operator, chart-operator, etc in the future.

@tfussell Thanks for driving this. It makes sense to me to have a separate repo to make the dependency graph more maintainable.

If you can apply this to cluster-apps-operator I think we can do the rest of the changes in honeybadger.

Just a couple of queries on docs and the readme.

File: README.md [hunk 1]
```diff
@@ -1,30 +1,3 @@
...
-5. Add this repository to https://github.com/giantswarm/github.
-6. Create quay.io docker repository if needed.
-7. Add the project to the CircleCI:
-   https://circleci.com/setup-project/gh/giantswarm/REPOSITORY_NAME
-8. Change the badge (with style=shield):
-   https://circleci.com/gh/giantswarm/REPOSITORY_NAME.svg?style=shield&circle-token=TOKEN_FOR_PRIVATE_REPO
-   If this is a private repository token with scope `status` will be needed.
-
-[generate]: https://github.com/giantswarm/template/generate
+# apiextensions-application
```
[the remark this answers is not in the document]
Comment (tfussell) [reply README.md:3]: Sure thing. I've added a good bit of info to the README now. Also noted in the CHANGELOG. Please take another look.

File: api/v1alpha1/app_catalog_entry_types.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/k8smetadata/pkg/annotation"
+)
+
+const (
+	kindAppCatalogEntry              = "AppCatalogEntry"
+	AppCatalogEntryDocumentationLink = "https://docs.giantswarm.io/ui-api/management-api/crd/appcatalogentries.application.giantswarm.io/"
```
[the remark this answers is not in the document]
Comment (tfussell) [reply api/v1alpha1/app_catalog_entry_types.go:11]: My plan was to modify the docs generator to know how to pull from multiple repos. Happy to prioritize it as soon as this set of changes is merged.

[the remark this answers is not in the document]
Comment (tfussell) [reply api/v1alpha1/app_catalog_entry_types.go:11]: The general idea would be to loop over a list of repositories in this function https://github.com/giantswarm/crd-docs-generator/blob/d8c24fcbec1523a4d522f195b35f04b314db4b02/main.go#L103

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rossf7) [reply README.md:3]: Nice! Readme looks good.

File: api/v1alpha1/app_catalog_entry_types.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (rossf7) [reply api/v1alpha1/app_catalog_entry_types.go:11]: Cool I wasn't sure how they were generated.

Supporting multiple repos will be needed but we don't have any CRD changes planned right now.

**Review by rossf7 — APPROVED** [review]
LGTM

**Pull request merged**